R/get_v1_data.R

```diff
@@ -273,28 +290,30 @@ spod_clean_zones_v1 <- function(zones_path) {
 #' # create a connection to the v1 data
 #' Sys.setenv(SPANISH_OD_DATA_DIR = "~/home/nosync/cache/mitma")
 #' dates <- c("2020-02-14", "2020-03-14", "2021-02-14", "2021-02-14", "2021-02-15")
-#' od_dist <- spod_get_od_v1(zones = "distr", dates = dates)
+#' od_dist <- spod_get_od(zones = "distr", dates = dates)
 #'
 #' # od dist is a table view filtered to the specified dates
 #'
 #' # access the source connection with all dates
 #' # list tables
 #' DBI::dbListTables(od_dist$src$con)
 #' }
-spod_get_od_v1 <- function(
+spod_get_od <- function(
     zones = c(
       "districts", "dist", "distr", "distritos",
       "municipalities", "muni", "municip", "municipios"
     ),
     dates = NULL,
     data_dir = spod_get_data_dir(),
     quiet = FALSE,
-    duck_max_mem = 2,
-    duck_max_threads = parallelly::availableCores())
+    duck_max_mem = 3,
```

Comment:
Why 3 here out of interest?

Reply:
@Robinlovelace tested with 2GB and it failed. I did not test it much beyond 7-10 files, we will see later how it behaves with more files or during conversion. In the HPC I'm used to giving it at least 30GB to convert the complete dataset. DuckDB is clever enough to [disk spilling](https://duckdb.org/2024/07/09/memory-management.html#:~:text=DuckDB%20deals%20with%20these%20scenarios%20by%20disk%20spilling) but in our case this 3GB seems like the bare minimum that it wants to be allocatable in RAM.

Reply:
OK makes sense. Agree at some point may be worth dynamically setting this.